src/datasets/inspect.py

```diff
@@ -118,13 +123,24 @@ def inspect_dataset(path: str, local_path: str, download_config: Optional[Downlo
         **download_kwargs (additional keyword arguments): Optional arguments for [`DownloadConfig`] which will override
             the attributes of `download_config` if supplied.
     """
-    dataset_module = dataset_module_factory(
-        path, download_config=download_config, force_local_path=local_path, **download_kwargs
-    )
+    dataset_module = dataset_module_factory(path, download_config=download_config, **download_kwargs)
+    builder_cls = import_main_class(dataset_module.module_path, dataset=True)
+    module_source_path = inspect.getsourcefile(builder_cls)
+    module_source_dirpath = os.path.dirname(module_source_path)
+    for dirpath, dirnames, filenames in os.walk(module_source_dirpath):
+        dst_dirpath = os.path.join(local_path, os.path.relpath(dirpath, module_source_dirpath))
+        os.makedirs(dst_dirpath, exist_ok=True)
+        for i, dirname in enumerate(dirnames[:]):
+            if dirname == "__pycache__":
+                del dirnames[i]
```

Comment:
Same here, and I guess you can ignore directories that start with "." and "__" to be more general, as we do in the download manager
```suggestion
        for i, dirname in enumerate(dirnames):
            if dirname.startswith((".", "__")):
                continue
```